src/Rule/Number.php

```diff
@@ -83,55 +89,116 @@ public function getName(): string
         return 'number';
     }
 
-    public function isAsInteger(): bool
+    /**
+     * Whether the value can only be an integer.
+     *
+     * @return bool Whether the value can only be an integer. Defaults to `false`.
+     *
+     * @see $integerOnly
+     */
+    public function isIntegerOnly(): bool
     {
-        return $this->asInteger;
+        return $this->integerOnly;
     }
 
+    /**
+     * Get lower limit of the number. `null` means no lower limit.
+     *
+     * @return float|int|null Lower limit of the number.
+     *
+     * @see $min
+     */
     public function getMin(): float|int|null
     {
         return $this->min;
     }
 
+    /**
+     * Get upper limit of the number. `null` means no upper limit.
+     *
+     * @return float|int|null Upper limit of the number.
+     *
+     * @see $max
+     */
     public function getMax(): float|int|null
     {
         return $this->max;
     }
 
+    /**
+     * Get error message used when the value is not numeric.
+     *
+     * @return string Error message.
+     *
+     * @see $incorrectInputMessage
+     */
     public function getIncorrectInputMessage(): string
     {
         return $this->incorrectInputMessage;
     }
 
+    /**
+     * Get error message used when the value is smaller than {@link $min}.
+     *
+     * @return string Error message.
+     *
+     * @see $tooSmallMessage
+     */
     public function getTooSmallMessage(): string
     {
         return $this->tooSmallMessage;
     }
 
+    /**
+     * Get error message used when the value is bigger than {@link $max}.
+     *
+     * @return string Error message.
+     *
+     * @see $tooBigMessage
+     */
     public function getTooBigMessage(): string
     {
         return $this->tooBigMessage;
     }
 
+    /**
+     * Get the regular expression for matching integers.
+     *
+     * @return string The regular expression.
+     *
+     * @see $integerPattern
+     */
     public function getIntegerPattern(): string
     {
         return $this->integerPattern;
     }
 
+    /**
+     * The regular expression for matching numbers.
+     *
+     * @return string The regular expression.
+     *
+     * @see $numberPattern
+     */
     public function getNumberPattern(): string
     {
         return $this->numberPattern;
     }
 
+    /**
+     * Get error message used when value type does not match.
+     *
+     * @return string Error message.
+     */
     public function getNotNumberMessage(): string
     {
-        return $this->asInteger ? 'Value must be an integer.' : 'Value must be a number.';
+        return $this->integerOnly ? 'Value must be an integer.' : 'Value must be a number.';
     }
 
     public function getOptions(): array
     {
         return [
-            'asInteger' => $this->asInteger,
+            'asInteger' => $this->integerOnly,
```

Comment:
option should be renamed if property is renamed, right?

Reply:
Yes.